apps/dashboard/src/components/primitives/permission-button.tsx

```diff
@@ -0,0 +1,69 @@
+import { Button, ButtonProps } from '@/components/primitives/button';
+import { PermissionsEnum } from '@novu/shared';
+import { ReactNode } from 'react';
+import { Tooltip, TooltipContent, TooltipTrigger } from './tooltip';
+import { useHasPermission } from '@/hooks/use-has-permission';
+
+export interface PermissionButtonProps extends ButtonProps {
+  /** The permission required to access this button functionality */
+  permission: PermissionsEnum;
+  /** Custom tooltip content to show when permission is denied (defaults to standard message) */
+  tooltipContent?: ReactNode;
+  /** Custom disabled button to show when permission is denied */
+  disabledButton?: ReactNode;
+  /** Custom permission check function (optional override for the default check) */
+  permissionCheck?: () => boolean;
+}
+
+export const PermissionButton = ({
+  permission,
+  tooltipContent,
+  children,
+  disabledButton,
+  permissionCheck,
+  mode,
+  asChild,
+  ...buttonProps
+}: PermissionButtonProps) => {
+  const has = useHasPermission();
+
+  const defaultPermissionCheck = () => has({ permission });
+  const canPerformAction = permissionCheck ? permissionCheck() : defaultPermissionCheck();
+
+  const defaultTooltipContent = (
+    <>
+      Almost there! Your role just doesn't have permission for this one.{' '}
+      <a href="https://docs.novu.co/" target="_blank" className="underline">
+        Learn More ↗
+      </a>
+    </>
+  );
+
+  if (!canPerformAction) {
```

Comment:
Should we feature flag the behaviour here? Or it doesn't matter because everyone will be Owners?

Reply:
The `has()` function is feature flag to return `true` if RBAC FF is not enabled, so this component will render as if you **can** perform the action = as it was before.